wallet/wallet.go

```diff
@@ -377,7 +384,42 @@ func (w *Wallet) syncWithChain() error {
 		}
 		ns := tx.ReadWriteBucket(waddrmgrNamespaceKey)
 
-		for height := int32(1); height <= bestHeight; height++ {
+		// Only allocate the recoveryMgr if we are actually in recovery
+		// mode.
+		var recoveryMgr *RecoveryManager
+		if isRecovery {
+			log.Infof("RECOVERY MODE ENABLED -- rescanning for "+
+				"used addresses with recovery_window=%d",
+				w.recoveryWindow)
+
+			// Initialize the recovery manager with a default batch
+			// size of 2000.
+			recoveryMgr = NewRecoveryManager(
+				w.recoveryWindow, 2000,
+			)
+
+			// In the event that this recovery is being resumed, we
+			// will need to repopulate all found addresses from the
+			// database. For basic recovery, we will only do so for
+			// the default scopes.
+			scopedMgrs, err := w.defaultScopeManagers()
+			if err != nil {
+				return err
+			}
+
+			txmgrNs := tx.ReadBucket(wtxmgrNamespaceKey)
```

Comment:
should check if bucket exists?

Reply:
I think we can assume it has been created, we check this in `Open` and initialize them in `Create`